File: Server/DriverServices/models/VehicleModel.js
import mongoose from "mongoose";

const Schema = mongoose.Schema;

const VehicleSchema = new Schema({
        color: {
            type: String,
            required: true,
        },
        licensePlate: {
            type: String,
            required: true,
        },
        image: {
            type: String,
            required: true,
        },
        typeId: {
            type: Schema.Types.ObjectId,
            required: true,
        },
    },
    {
        timestamps: true,
    }
);

const MotorbikeSchema = new Schema(VehicleSchema);

const CarSchema = new Schema(VehicleSchema);
CarSchema.add({
    capacity: {
        type: Number,
        required: true,
    },
});

export const Vehicle = mongoose.model('Vehicle', VehicleSchema, 'vehicles');
export const Motorbike = Vehicle.discriminator('Motorbike', MotorbikeSchema);
export const Car = Vehicle.discriminator('Car', CarSchema);

File: Client/App/Driver/lib/global/models/vehicle/vehicle_model.dart
class VehicleModel {
  final String color;
  final String licensePlate;
  final String image;
  final String typeId;
  
  VehicleModel({
    required this.color,
    required this.licensePlate,
    required this.image,
    required this.typeId,
  });

  factory VehicleModel.fromJson(Map<String, dynamic> json) {
    return VehicleModel(
      color: json['color'],
      licensePlate: json['licensePlate'],
      image: json['image'],
      typeId: json['typeId'],
    );
  }
}
